mvlearn/decomposition/groupica.py

```diff
@@ -0,0 +1,326 @@
+        self.ica_kwargs = ica_kwargs
+        self.random_state = random_state
+
+    def _fit(self, Xs, y=None):
+        """
+        Fit  to the data and transform the data
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        sources : array, shape (n_samples, n_components)
+             Estimated sources
+        """
+        Xs = check_Xs(Xs, copy=True)
+        self.means_ = [np.mean(X, axis=0) for X in Xs]
+        gpca = GroupPCA(
+            n_components=self.n_components,
+            n_individual_components=self.n_individual_components,
+            copy=True,
+            prewhiten=self.prewhiten,
+            whiten=True,
+            random_state=self.random_state,
+        )
+        X_pca = gpca.fit_transform(Xs)
+        self.grouppca_ = gpca
+        if self.solver == "fastica":
+            K, W, sources = fastica(
+                X_pca, **self.ica_kwargs, random_state=self.random_state
+            )
+        else:
+            K, W, sources = picard(
+                X_pca.T, **self.ica_kwargs, random_state=self.random_state
+            )
+            sources = sources.T
+        if K is not None:
+            self.components_ = np.dot(W, K)
+        else:
+            self.components_ = W
+        self.mixing_ = linalg.pinv(self.components_)
+        # Compute individual unmixing matrices by least-squares
+        self.individual_mixing_ = []
+        self.individual_components_ = []
+        sources_pinv = linalg.pinv(sources)
+        for X, mean in zip(Xs, self.means_):
+            lstq_solution = np.dot(sources_pinv, X - mean)
+            self.individual_components_.append(linalg.pinv(lstq_solution).T)
+            self.individual_mixing_.append(lstq_solution.T)
+        self.n_components_ = gpca.n_components_
+        self.n_features_ = gpca.n_features_
+        self.n_samples_ = gpca.n_samples_
+        self.n_subjects_ = gpca.n_subjects_
+        return sources
+
+    def fit_transform(self, Xs, y=None):
+        """
+        Fit to the data and transform the data into sources
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        X_transformed : list of array-likes or numpy.ndarray
+            The transformed data.
+            If `multiple_outputs` is True, it is a list with the estimated
+            individual sources.
+            If `multiple_outputs` is False, it is a single array containing the
+            shared sources.
+        """
+        sources = self._fit(Xs, y)
+        if self.multiple_outputs:
+            return self.transform(Xs)
```

Comment:
fit_transform must be the same as fit().transform

Reply:
It is the same so far, the `test_transform` function in `tests/decomposition/test_groupica.py` makes sure that is behaves like this. 

I thought that this was the simplest way to not have duplicate code: the `_fit` function returns the shared sources, which is the expected output of `fit_transform` is `multiple_outputs` is False. Otherwise, we still need to compute the individual sources.

Reply:
I don't get why you need to branch here. transform should already take multiple_outputs param into account

Reply:
It's purely in order to avoid doing the same computation twice:

If multiple_outputs = False, you already have computed the fit_transform in the fitting process. If multiple_outputs = True, then you have some more work to do.

Reply:
don't have time to look closer but it's suspicious to me. Also you branch in transform too...

Reply:
Well I don't know if there is a more readable way to do, but all tests pass, so it has the expected behavior, and I think that this way of doing avoids unnecessary computations.